xlators/protocol/server/src/server-resolve.c

```diff
@@ -100,6 +100,45 @@ resolve_gfid_entry_cbk(call_frame_t *frame, void *cookie, xlator_t *this,
     return 0;
 }
 
+int
+resolve_name(call_frame_t *frame, inode_t *parent)
+{
+    server_state_t *state = NULL;
+    server_resolve_t *resolve = NULL;
+    loc_t *resolve_loc = NULL;
+    dict_t *dict = NULL;
+    xlator_t *this = frame->this;
+
+    state = CALL_STATE(frame);
+    resolve = state->resolve_now;
+    resolve_loc = &resolve->resolve_loc;
+    resolve_loc->parent = inode_ref(parent);
+    gf_uuid_copy(resolve_loc->pargfid, resolve_loc->parent->gfid);
+
+    resolve_loc->name = resolve->bname;
+
+    resolve_loc->inode = server_inode_new(state->itable, resolve_loc->gfid);
+
+    inode_path(resolve_loc->parent, resolve_loc->name,
+               (char **)&resolve_loc->path);
+
+    if (state->xdata) {
+        dict = dict_copy_with_ref(state->xdata, NULL);
+        if (!dict)
+            gf_msg(this->name, GF_LOG_ERROR, ENOMEM, PS_MSG_NO_MEMORY,
+                   "BUG: dict allocation failed (pargfid: %s, name: %s), "
```

Comment:
Why is it a BUG?

Reply:
This was just a re-arrangement of old code. I am not completely sure why it is a BUG. @amarts Do you know?

Reply:
Yes, it shouldn't be a 'BUG'. I guess, instead of just mentioning log level as ERROR, this 'BUG:' string is added.

Shouldn't impact this patch, but there are overall 5 `BUG: ` in logs in codebase, we can look at all 5 and fix them.

Reply:
I went through this code path, it doesn't look like it is easy to set an errno if there are any failures in this code path. May have to introduce this to handle errors correctly. I will work on it separately when I get some time. I raised https://github.com/gluster/glusterfs/issues/2628 for this purpose